File: pset1/mario/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    int size = 0;

    do
    {

        // get the size
        size = get_int("Height: ");

        // check if the size falls under the specified range
        if (size >= 1 && size <= 8)
        {
            break;
        }

    }
    while (true);

    int spaces = size - 1;

    for (int i = 0; i < size; i++)
    {
        // Print the spaces
        for (int j = spaces; j > 0; j--)
        {
            printf(" ");
        }

        // print the hashes
        for (int k = 0; k <= i; k++)
        {
            printf("#");
        }

        // print the empty 2 spaces
        printf("  ");

        // print the oposite pyramid
        for (int j = 0; j <= i; j++)
        {
            printf("#");
        }

        printf("\n");
        spaces -= 1;
    }
}

File: pset6/mario/more/mario.py
from cs50 import get_int


def get_input():
    while True:
        size = get_int("Height: ")
        if size >= 1 and size <= 8:
            break

    return size


def make_pyramid(height):
    spaces = height - 1

    while spaces >= 0:
        print(" " * spaces, end="")
        print("#" * (height - spaces), end="")
        print("  ", end="")
        print("#" * (height - spaces), end="")
        print()

        spaces -= 1

size = get_input()
make_pyramid(size)
